### ISSUE TITLE
T5/trainer_t5.py validation

### ISSUE THREAD

**Issue by Sejik** [glee4810/EHRSQL#5]
valid_loss = model(
                               input_ids=input_ids,
                               labels=labels)[0]
valid_loss_list.append(valid_loss.item())

I think we need valid_loss = torch.mean(valid_loss).

valid_loss = model(
                               input_ids=input_ids,
                               labels=labels)[0]
valid_loss = torch.mean(valid_loss).
valid_loss_list.append(valid_loss.item())

**Comment by Sejik** [comment]
or 
valid_loss_list.extend(i.item() for i in valid_loss)

**Comment by Sejik** [comment]
Also, for args.show_eval_sample,
"dataparallel makes the error" which might be solved by using model.module.generate. (instead of model.generate)

**Comment by glee4810** [comment]
Seems like you are running the model using dataparallel. You are right if you want to use multiple GPUs to train the model.

**Comment by Sejik** [comment]
Thank you.
